lodash.src.js

```diff
@@ -9777,6 +9777,44 @@
     }
 
     /**
+     * Resolves the value of `keyPath` on `object`. If any values along `keyPath`
+     * is a function it is invoked with the `this` binding of `object` and its
+     * result is used to resolve the remainder of the `keyPath` rather than the
+     * function itself. If at any point along the `keyPath` the value is
+     * `undefined` the `defaultValue` is returned.
+     *
+     * @static
+     * @memberOf _
+     * @category Object
+     * @param {Object} object The object to query.
+     * @param {array} keyPath The key path of the properties to resolve.
+     * @param {*} [defaultValue] The value returned if the property value
+     *  resolves to `undefined`.
+     * @returns {*} Returns the resolved value.
+     * @example
+     *
+     * var object = {
+     *  'user': 'fred',
+     *  'dates': { 'birthdate': '03/17/1988', anniversary: '07/29/2012' }
+     * };
+     *
+     * _.result(object, ['user']);
+     * // => 'fred'
+     *
+     * _.result(object, ['user', 'dates', 'anniversary']);
+     * // => '07/29/2012'
+     *
+     * _.result(object, ['status', 'timestamp'], _.constant('busy'));
+     * // => 'busy'
+     */
+    function resultDeep(object, keyPath, defaultValue) {
+      arrayEach(dropRight(keyPath), function(key) {
+        return object = result(object, key);
```

Comment:
Correct, this is intentional to avoid excessive looping.

How about taking it a step further and simply casting the result to a boolean? I can't think of any falsey values that have enumerable properties, so it seems safe to me (albeit a bit cryptic for passersby).
```javascript
return !!(object = result(object, key));
```